### DISCUSSION TITLE
Accordion

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#217] [Component suggestions]
<img width="480" alt="visuele weergave van het accordion component" src="https://github.com/nl-design-system/backlog/assets/545207/d17d58a3-72c1-40d1-9ed2-9521226ccf24">

We willen graag meer leren over het **Accordion** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1707406585876149)

---

## Naam
Accordion

## Doel
Secties met de mogelijkheid om gerelateerde content te tonen en weer te verbergen.

## Gerelateerde componenten
[Details](https://github.com/orgs/nl-design-system/discussions/218), [Tabs](https://github.com/orgs/nl-design-system/discussions/232)

## Onderzoek
[Accordions on desktop - Nielsen Norman Group](https://www.nngroup.com/articles/accordions-on-desktop/)
Artikel waarin wordt beschreven wat de voor- en nadelen zijn van een accordion en waar je rekening mee moet houden.

[Accordions Are Not Always the Answer for Complex Content on Desktops - Nielsen Norman Group](https://www.nngroup.com/articles/accordions-complex-content/)
Artikel waarin wordt beschreven dat Accordions niet altijd de juiste oplossing bieden.

[Accordions icons - Nielsen Norman Group](https://www.nngroup.com/articles/accordion-icons/)
Artikel waarin het onderzoek naar iconen voor Accordions wordt beschreven. Helaas is positie niet meegenomen en betreffende de getoonde voorbeelden ook veelal navigatie / menu's.

[Accordion - GOV.UK](https://design-system.service.gov.uk/components/accordion/)
Onder 'Research on this component' worden bevindingen genoteerd. Er is ook een [link naar de Accordion Github discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/1) waar meer informatie, onderzoek en aanpassingen worden gedeeld.

[Collapsible sections - Inclusive components](https://inclusive-components.design/collapsible-sections/)
Artikel waarin een Accordion component uitvoerig wordt beschreven om zo toegankelijk op te zetten.

[Don't be afraid of the big long page](https://amyhupe.co.uk/articles/dont-be-afraid-of-the-big-long-page/)
Artikel waarin wordt beschreven dat het inzetten van Accordions (of tabs) niet altijd de oplossing zijn.

**Comment by MikeClickr** [comment]
@jeffreylauwers In het Amsterdam Design System staat een beschrijving: https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=2927%3A29173&mode=design&t=SfVgJtnxdqw3HzMQ-1
In het Erfpachtportaal dat Nico Druif onlangs presenteerde, zijn 'Toon' en 'Verberg' links toegepast, dat werd niet door iedereen begrepen. Ik vermoed dat chevrons voldoende zijn, mits deze visueel verbonden zijn.

Bij mijn vorige werkgever heb ik ook accordeons voor de G4 ontworpen, een soort G4 Design System met @hidde onder meer.
Elke gemeente had toen zo'n accordeon component, zie een screenshot: https://informaat.nl/nl/cases/wigo4it
Ik heb de gebruikstesten niet meer, maar ze werden goed gebruikt herinner ik me.

**Comment by Rozerinay** [comment]
Ik heb 2 inzichten gevonden uit ons onderzoek naar denhaag.nl in 2022.

- Accordeons op de productpagina's zorgen ervoor dat de website makkelijker te gebruiken zijn. Sommigen vinden de hoeveelheid tekst op de productpagina's overweldigend op de desktopversie. De accordeons op de mobiele versie maken deze tekst beter hanteerbaar.
- Groene knoppen binnen accordeons werden wel gevonden, maar ook als verwarrend ervaren. Respondenten verwachten een knop of helemaal onderaan of bovenaan op de pagina.

Als je meer wilt weten over dit onderzoek, bekijk dan het Miro bord [DH.nl Onderzoek ronde 2](https://miro.com/app/board/uXjVPYSokU0=/?moveToWidget=3458764534078382780&cot=14) voor gedetailleerde inzichten en bevindingen.

**Comment by Rozerinay** [comment]
Verder hebben we onlangs onderzoek gedaan naar de formulieren. In dat onderzoek is de productpagina ook onderzocht waar de accordion in voorkwam. Het inzicht dat we daaruit hebben gehaald is:

De accordion bij 'Dit heeft u nodig' maakt het overzicht scanbaar. Als alles gesloten is in de accordion dan zie je alle documenten die je nodig hebt.

<img width="796" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/b9d343be-0161-4ff2-8428-eed001c7fe39">

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-containers-accordion--docs)
- [GitHub](https://github.com/Amsterdam/design-system/blob/develop/packages/css/src/components/accordion)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=4442-26719&mode=design&t=yypc3EdrKU17akla-4)

---

### Screenshots / Varianten
**Storybook**
![Scherm­afbeelding 2024-02-12 om 12 12 46](https://github.com/nl-design-system/backlog/assets/545207/83467f84-7a9b-4834-98eb-07835fd720d4)

**Figma**
<img width="490" alt="Scherm­afbeelding 2024-02-12 om 12 13 38" src="https://github.com/nl-design-system/backlog/assets/545207/bc37a181-590e-4893-8b29-2b8686ce0ba5">

---

### Notities
- Verschil tussen Figma en Storybook. In Figma border om accordion section.
- Richtlijnen en beschrijving van keyboard navigatie beschikbaar (Storybook).

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-surfaces-accordion--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/blob/main/components/Accordion)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1585-0&mode=design&t=GZk2Yw0oc0iJwAyF-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=1261-4785&mode=design&t=0fP4N8iuiAt7pNmx-4)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-02-12 om 12 29 45](https://github.com/nl-design-system/backlog/assets/545207/1b419352-7e4f-47f1-9621-933adbedde00)

**Figma WIP**
![Scherm­afbeelding 2024-03-06 om 13 29 09](https://github.com/nl-design-system/backlog/assets/545207/05473baa-28c1-4c2a-bc44-bed807dc34a9)




---

### Notities
-  Richtlijnen beschikbaar.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

- [Documentatie website](https://dso-toolkit.nl/62.8.4/components/navigatie/accordion)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/accordion)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-02-12 om 12 44 39](https://github.com/nl-design-system/backlog/assets/545207/f2a0a5de-f30e-40c9-91bd-914b4aeb7ae9)

Compact
![Scherm­afbeelding 2024-02-12 om 12 45 14](https://github.com/nl-design-system/backlog/assets/545207/da281ab8-b33f-427b-9b87-4932fb056d18)

Conclusie
![Scherm­afbeelding 2024-02-12 om 12 45 44](https://github.com/nl-design-system/backlog/assets/545207/4dd12f4d-bd80-49d9-8219-d2eb842d68f8)

---

### Notities
- Richtlijnen beschikbaar.
- 3 varianten, geen richtlijnen gevonden over wanneer welke variant te gebruiken.
- Voor de Default variant zouden de volgende design tokens voor Candidate nodig zijn:
  - `nl.accordion.button.expanded.background-color`
  - `nl.accordion.button.expanded.color`

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/collapse)

---

### Screenshots / Varianten

**Documentatie website**

![Scherm­afbeelding 2024-02-12 om 12 51 30](https://github.com/nl-design-system/backlog/assets/545207/a9b923c3-c05a-4bb7-85af-6141a515d9a8)


---

### Notities
- Onder de naam: Collapse
- Ook varianten voor toepassing binnen Table en zonder achtergrond.
- Accordion-ish element als onderdeel van [Video](https://uno.dfront.rijkscloud.nl/#/components/video) component.

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

- [Documentatie website - Accordeon](https://componenten.lenteveld.nl/accordeon)
- [Documentatie website - Pagina accordeon](https://componenten.lenteveld.nl/pagina-accordeon)
- [Veenendaal -Collapsible in productie](https://www.veenendaal.nl/inwonerszaken/identiteit/rijbewijs-aanvragen-en-ophalen)
- [Patternlab - Accordion](https://patternlab.lenteveld.nl/patterns/02-organisms-010-Accordions/index.html)

---

### Screenshots / Varianten

**Accordeon**
De contentelement ‘accordeon’ gebruiken we op een homepage om zoveel mogelijk onderwerpen gegroepeerd te tonen, in twee klikken naar je bestemming (als het even kan).
![Scherm­afbeelding 2024-02-12 om 13 02 17](https://github.com/nl-design-system/backlog/assets/545207/5947ad0f-665e-4514-a769-dd180edfaf89)
![Scherm­afbeelding 2024-02-12 om 13 02 05](https://github.com/nl-design-system/backlog/assets/545207/0ddee149-8b4d-4847-8a62-78448a939121)

**Collapsible**
Dit is in TYPO3 het tekst en media contentelement dat ingesteld kan worden als collapsible. Eigenlijk een accordeon voor product pagina’s. Wij adviseren webredacteuren om deze alleen te gebruiken als er specifieke informatie in wordt getoond die niet voor iedereen relevant is omdat je met een collapsible informatie verbergt en dat is eigenlijk niet goed.

![Scherm­afbeelding 2024-02-12 om 13 08 26](https://github.com/nl-design-system/backlog/assets/545207/1877fabe-dd5b-455e-be78-696d914e1330)

---

### Notities
- Twee soorten accordeons.
- Bij Accordeon kan maximaal 1 section open zijn. Bij Collapsible kunnen meerdere sections open zijn.
- Scroll positie wordt aangepast na openklikken section.
- Accordion-ish element als onderdeel van [Mediaplayer](https://componenten.lenteveld.nl/mediaplayer) component.

**Comment by jeffreylauwers** [comment]
## Rotterdam
[Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=20033-12056&mode=design&t=BMOuTBbnw8xr36Vf-4)

---

### Screenshots / Varianten

**Figma**

Icon false
<img width="444" alt="Scherm­afbeelding 2024-02-12 om 13 23 11" src="https://github.com/nl-design-system/backlog/assets/545207/0ccfe6ff-0ccc-4249-9a74-9b01447fbc35">

Icon true
<img width="442" alt="Scherm­afbeelding 2024-02-12 om 13 23 18" src="https://github.com/nl-design-system/backlog/assets/545207/981ca548-3174-48ac-a5ca-7e237c2de621">


---

### Notities
- Expliciet benoemd dat er 'no rules' zijn wat betreft de inhoud van de sections.
- Accordion-ish element voor [FAQ component](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=543-27142&mode=design&t=mt45Bpho3t3nAR2w-4) in Figma.

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-accordion--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/accordion)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=5207-16178&mode=design&t=rglklLUpYik7Vr5b-4)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-02-12 om 13 28 21](https://github.com/nl-design-system/backlog/assets/545207/a8d50c7f-a2c3-469a-9e6c-d853b27c7c69)

---

### Notities
- Teaser text per section.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-accordion--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/blob/main/components/accordion)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=1261-4785&mode=design&t=TbWAXe8TEdVvngTx-4)

---

### Screenshots / Varianten

**Figma**
<img width="500" height="260" alt="Scherm­afbeelding 2025-08-06 om 13 17 01" src="https://github.com/user-attachments/assets/8b618cd2-267f-4e16-8735-9c679a8a5b0a" />

**Storybook**
<img width="967" height="217" alt="Scherm­afbeelding 2025-08-06 om 13 16 10" src="https://github.com/user-attachments/assets/8166826c-b295-45b2-91e7-9bdaaef1792b" />


---

### Notities
- Chevron stond eerder in een geel blokje rechts uitgelijnd. Daar wil Utrecht van afstappen en het icoon links uitlijnen, vandaar enkel screenshot Figma omgeving.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Accordion op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=accordion)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Accordion op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=accordion)

**Comment by MarjonBakker** [comment]
## Alles inklappen/uitklappen

Vanuit toegankelijkheid heb ik een vraag over dit component. Ik zie in de discussie hierboven dat er nergens een button zit met een functie 'Alles openklappen'/'Alles dichtklappen'. Het toevoegen van zo'n button heeft meerdere voordelen, bijvoorbeeld:

- Mensen met een motorische beperking hoeven niet meerdere buttons te activeren om alle content te kunnen bekijken.
- Als je content op een pagina wilt vinden met de zoekfunctie van je browser, lukt dat het beste met alle content uitgeklapt.

GOV.UK heeft deze button wel toegevoegd: https://design-system.service.gov.uk/components/accordion/

Is het een overweging waard om dat hier ook toe te voegen?

**Comment by julezrulez** [comment]
Een groot probleem met het aanbieden van uitvouwmechanismen  is dat je dit alleen zou moeten aanbieden op de minder relevante informatie. Door het component aan webredacties aan te bieden veroorzaak je een hoop leed bij de bezoekers van de website. Als je een officieel document wilt aanvragen wil je graag weten wat je moet doen, wanneer, de kosten en wat je mee moet nemen. Je ziet die informatie nu vaak in verschillende uitvouwmechanismen staan. Dat betekent dat je elk onderdeel open moet klikken. Ik heb zelfs een voorbeeld gezien waar je een zo genaamde accordion opende voor informatie waar vervolgens slechts één linkje wordt aangeboden. Nog erger zijn de uitvouwdingen waarbij je de voor jou relevante uitvouwdingen opent en vervolgens blijken de andere allemaal gesloten. Nee, dat is overzichtelijk; NOT! Nog erger wordt het als je voor je ouders of grootouders de boel even wilt printen omdat oudere mensen stomweg weigeren te internet in huis te halen. Dan kom je met je printje en blijken er nog belangrijke informatie dichtgeklapt te zijn. Tip voor developers: uitklapmechanismen werken niet op papier.

Tot slot de reden waarom ik hier ben; ik was op zoek naar een goede toepassing van details/summary. Maar mis ik die oplossing hier nou? Dat zou toch de standaard methode moeten zijn om een pagina overzichtelijker te maken?

**Reply by jeffreylauwers** [reply]
Bedankt voor je opmerkingen Julez. Dit zijn allemaal waardevolle punten die we kunnen meenemen als we de documentatie van de Accordion component verder uitwerken. We willen hier dan ook meegeven dat men niet zomaar naar de Accordion moet grijpen. Zie ook [het artikel van Amy Hupe](https://amyhupe.co.uk/articles/dont-be-afraid-of-the-big-long-page/) bij ‘Onderzoek’ bovenaan deze discussion.

Voor het printen zouden we developers de richtlijn kunnen meegeven om [CSS te schrijven die specifiek voor printen word uitgevoerd](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_media_queries/Printing), waarin alle uitvouwelementen standaard uitgevouwd zijn.

Ik heb begrepen dat  details/summary ook werkt met de browser zoekfunctie. Als je dus zoekt op “Something”, en dat komt voor binnen de details/summary, dan vind de browser het en opent de browser het wanneer je door de zoekresultaten loopt. 

Als designer ben ik even in de code gaan kijken van bovenstaande verzameling community componenten. [DUO](https://uno.dfront.rijkscloud.nl/#/components/collapse) en [RVO](https://nl-design-system.github.io/rvo/?path=/docs/componenten-accordion--documentatie) lijken details/summary te gebruiken voor hun accordion. Helpt dat?

We zijn trouwens een aparte [GitHub Discussion gestart voor de Details component](https://github.com/orgs/nl-design-system/discussions/218). Zelf ben ik wel enthousiast over de implementatie van RVO. Hoe denk jij daar over?

**Comment by julezrulez** [comment]
Ik kan bij RVO er helaas nog geen goedkeuring aan geven. De naam van het eerste uitvouwmechanisme luidt: "Delta omlaag Accordion item 1 Teaser for accordion item 1". Weet een bezoeker die niks van front-end weet nu wat hiermee bedoeld wordt: "Delta", "Teaser", "Accordion"???  De "front-end"-developer maakt het hier zichzelf veel te lastig; veelsteveel overbodige code. Nee helaas, dit is geen productiewaardig materiaal; dit moet je eerst door een toegankelijkheidsexpert laten controleren voor dit live mag.
Ik ben daarom enthousiaster van de DUO-oplossing.
@jeffreylauwers ik ben benieuwd waarom je de voorkeur hebt voor de oplossing van RVO, ten opzichte van DUO.

**Reply by jeffreylauwers** [reply]
Ah dat is mooie feedback op dit component. Ik ga het doorgeven aan RVO. Is de Accordion bij DUO qua code goed opgezet? Dan kan ik daar direct naar verwijzen.

Oh ik heb geen voorkeur voor de RVO danwel DUO variant van de Details component. Ik was vooral benieuwd of die implementatie qua toegankelijkheid goed in elkaar stak. Of is daar hetzelfde euvel als bij de Accordion?

**Reply by julezrulez** [reply]
Ik ga geen uitspraken doen of iets goed is geïmplementeerd. Ik heb geen controle over de design systemen en ben er ook niet bij betrokken. Wat ik nu beoordeel is morgen misschien al weer gewijzigd. Wat mooi zou zijn is dat je een stempel kan zetten op die componenten die goed beoordeeld zijn, waarbij het stempel verdwijnt zodra wijzigingen in de code hebben plaatsgevonden. Ook zou je componenten het kenmerk "voldoet niet" moeten kunnen geven om zo het verspreiden van toegankelijkheidsproblemen te stoppen. Het overnemen van goede voorbeelden is nu te veel gebaseerd op assumptie. En assumptie is de grootste vijand van toegankelijkheid.
